packages/ckeditor5-list/src/documentlist/documentlistediting.js

```diff
@@ -455,6 +457,50 @@ export default class DocumentListEditing extends Plugin {
 			evt.return = fixListItemIds( listNodes, seenIds, writer ) || evt.return;
 		}, { priority: 'high' } );
 	}
+
+	/**
+	 * Integrates the feature with the clipboard via {@link module:engine/model/model~Model#getSelectedContent}.
+	 *
+	 * @private
+	 */
+	_setupClipboardIntegration() {
+		const model = this.editor.model;
+
+		// To enhance the UX, the editor should not copy list attributes to the clipboard if the selection
+		// started and ended in the same list item block or a block object as a list block was selected.
+		//
+		// * This prevents splitting list items when such a content is pasted/dropped by the user (block widgets in particular),
+		//   * Clipboard integration heuristics of ImageInlineEditing can kick in.
+		// * This avoids pasting single-list-item lists (orphans) anywhere else (in the root, in a paragraph).
+		//
+		//	                       ┌─────────────────────┬───────────────────┐
+		//	                       │ Selection           │ Clipboard content │
+		//	                       ├─────────────────────┼───────────────────┤
+		//	                       │ * [<blockObject />] │ <blockObject />   │
+		//	                       ├─────────────────────┼───────────────────┤
+		//	                       │ * Foo [bar] baz     │ bar               │
```

Comment:
I guess I over-engineered this bit. I don't think there's any way for this kind of selection (even if in spans the entire block) to include list attributes in `getSelectedContent()`. Or at least I couldn't reproduce this. If my concerns are confirmed, I'll simplify this code because it looks like it only applies to block object only after all.